#### Se realizará la carga de librerías para analizar el data set

In [5]:
# Cargaremos las siguientes librerías para hacer una revisión inicial del dataset

import pandas as pd


In [10]:
df = pd.read_csv("C:/Users/kurtc/Documents/GitHub/Proyecto-Final/Data/Student_performance_data.csv", index_col="StudentID")

In [8]:
# Se hará revisión de la información que contiene el dataset

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2392 entries, 0 to 2391
Data columns (total 15 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   StudentID          2392 non-null   int64  
 1   Age                2392 non-null   int64  
 2   Gender             2392 non-null   int64  
 3   Ethnicity          2392 non-null   int64  
 4   ParentalEducation  2392 non-null   int64  
 5   StudyTimeWeekly    2392 non-null   float64
 6   Absences           2392 non-null   int64  
 7   Tutoring           2392 non-null   int64  
 8   ParentalSupport    2392 non-null   int64  
 9   Extracurricular    2392 non-null   int64  
 10  Sports             2392 non-null   int64  
 11  Music              2392 non-null   int64  
 12  Volunteering       2392 non-null   int64  
 13  GPA                2392 non-null   float64
 14  GradeClass         2392 non-null   float64
dtypes: float64(3), int64(12)
memory usage: 280.4 KB


#### Podemos ver que al haber utilizado ".info" no existen datos vaciós en el dataset

In [11]:
df
# Al ver que la base contiene un identificador para los estudiantes, quitaremos el index default de Python y dejaremos el que contiene el dataset

Age  Gender  Ethnicity  ParentalEducation  StudyTimeWeekly  \
StudentID                                                               
1001        17       1          0                  2        19.833723   
1002        18       0          0                  1        15.408756   
1003        15       0          2                  3         4.210570   
1004        17       1          0                  3        10.028829   
1005        17       1          0                  2         4.672495   
...        ...     ...        ...                ...              ...   
3388        18       1          0                  3        10.680555   
3389        17       0          0                  1         7.583217   
3390        16       1          0                  2         6.805500   
3391        16       1          1                  0        12.416653   
3392        16       1          0                  2        17.819907   

           Absences  Tutoring  ParentalSupport  Extracurricular  Sports  \
StudentID                                                                 
1001              7         1                2                0       0   
1002              0         0                1                0       0   
1003             26         0                2                0       0   
1004             14         0                3                1       0   
1005             17         1                3                0       0   
...             ...       ...              ...              ...     ...   
3388              2         0                4                1       0   
3389              4         1                4                0       1   
3390             20         0                2                0       0   
3391             17         0                2                0       1   
3392             13         0                2                0       0   

           Music  Volunteering       GPA  GradeClass  
StudentID                                             
1001           1             0  2.929196         2.0  
1002           0             0  3.042915         1.0  
1003           0             0  0.112602         4.0  
1004           0             0  2.054218         3.0  
1005           0             0  1.288061         4.0  
...          ...           ...       ...         ...  
3388           0             0  3.455509         0.0  
3389           0             0  3.279150         4.0  
3390           0             1  1.142333         2.0  
3391           1             0  1.803297         1.0  
3392           0             1  2.140014         1.0  

[2392 rows x 14 columns]

In [12]:
df.shape

(2392, 14)

In [14]:
df.describe()

Age       Gender    Ethnicity  ParentalEducation  \
count  2392.000000  2392.000000  2392.000000        2392.000000   
mean     16.468645     0.510870     0.877508           1.746237   
std       1.123798     0.499986     1.028476           1.000411   
min      15.000000     0.000000     0.000000           0.000000   
25%      15.000000     0.000000     0.000000           1.000000   
50%      16.000000     1.000000     0.000000           2.000000   
75%      17.000000     1.000000     2.000000           2.000000   
max      18.000000     1.000000     3.000000           4.000000   

       StudyTimeWeekly     Absences     Tutoring  ParentalSupport  \
count      2392.000000  2392.000000  2392.000000      2392.000000   
mean          9.771992    14.541388     0.301421         2.122074   
std           5.652774     8.467417     0.458971         1.122813   
min           0.001057     0.000000     0.000000         0.000000   
25%           5.043079     7.000000     0.000000         1.000000   
50%           9.705363    15.000000     0.000000         2.000000   
75%          14.408410    22.000000     1.000000         3.000000   
max          19.978094    29.000000     1.000000         4.000000   

       Extracurricular       Sports        Music  Volunteering          GPA  \
count      2392.000000  2392.000000  2392.000000   2392.000000  2392.000000   
mean          0.383361     0.303512     0.196906      0.157191     1.906186   
std           0.486307     0.459870     0.397744      0.364057     0.915156   
min           0.000000     0.000000     0.000000      0.000000     0.000000   
25%           0.000000     0.000000     0.000000      0.000000     1.174803   
50%           0.000000     0.000000     0.000000      0.000000     1.893393   
75%           1.000000     1.000000     0.000000      0.000000     2.622216   
max           1.000000     1.000000     1.000000      1.000000     4.000000   

        GradeClass  
count  2392.000000  
mean      2.983696  
std       1.233908  
min       0.000000  
25%       2.000000  
50%       4.000000  
75%       4.000000  
max       4.000000

### Ahora haremos una descripción de las diferentes variables dentro del dataset
- 0 -  **StudentID (int)**: ID del estudiante (desde el 1001 al 2392)
- 1 - **Age (int)**: Edad del estudiante
- 2 - **Gender (int)**: Género del estudiante  
- 3 - **Ethnicity (int)**: Etnia del estudiante donde 0 = Caucasian, 1 = African American, 2 = Asian, 3 = Other
- 4 - **ParentalEducation (int)**: Nivel de educación de los padres donde 0 = none, 1 = High School, 2 = Some College, 3: Bachelor, 4 = Higher
- 5 - **StudyTimeWeekly (float)**: Tiempo de estudio semanal en horas
- 6 - **Absences (int)**: Ausentismo durante el año escolar
- 7 - **Tutoring (int)**: Tiene tutorías, donde 0 = no y 1 = si
- 8 - **ParentalSupport (int)**: Nivel de apoyo de padres, el cual es clasificado de 0 a 4, siendo 0 = Nada y 4 = Muy alto
- 9 - **Extracurricular (int)**: Actividades extracurriculares
- 10 - **Sports (int)**: Participa o no en deportes donde 0 = no y 1 = sí
- 11 - **Music (int)**: Participa o no en música donde 0 = no y 1 = sí
- 12 - **Volunteering (int)**: Participa o no en voluntariados donde 0 = no y 1 = sí
- 13 - **GPA (float)**: Promedio de calificaciones en una escala de 2.0 a 4.0, influenciado por los hábitos de estudio, la participación de los padres y las actividades extracurriculares.
- 14 - **GradeClass (float)**: Clasificación de las calificaciones de los estudiantes según el GPA